Sources/PrimerSDK/Classes/Core/3DS/ThreeDSecureService.swift

```diff
@@ -0,0 +1,207 @@
+//
+//  ThreeDService.swift
+//  PrimerSDK
+//
+//  Created by Evangelos Pittas on 1/4/21.
+//
+
+#if canImport(UIKit)
+
+import Foundation
+import ThreeDS_SDK
+
+protocol ThreeDSecureServiceProtocol {
+    func threeDSecureBeginAuthentication(paymentMethodToken: PaymentMethodToken,
+                                         threeDSecureBeginAuthRequest: ThreeDS.BeginAuthRequest,
+                                         completion: @escaping (ThreeDS.BeginAuthResponse?, Error?) -> Void)
+}
+
+class ThreeDSecureService: ThreeDSecureServiceProtocol {
+    
+    @Dependency private(set) var state: AppStateProtocol
+    @Dependency private(set) var api: PrimerAPIClientProtocol
+    
+    let threeDS2Service: ThreeDS_SDK.ThreeDS2Service = ThreeDS2ServiceSDK(bundle: Bundle.primerFramework)
+    private var transaction: Transaction?
+    private var netceteraCompletion: ((_ netceteraThreeDSCompletion: ThreeDS.NetceteraThreeDSCompletion?, _ err: Error?) -> Void)?
+    
+    deinit {
+        print("ThreeDSecureServiceProtocol deinit")
+    }
+    
+    func initializeSDK(completion: @escaping (Result<Void, Error>) -> Void) {
+        do {
+            let configBuilder = ConfigurationBuilder()
+            try configBuilder.log(to: .debug)
+            try configBuilder.license(key: Primer.netceteraLicenseKey)
+            
+            let supportedSchemeIds: [String] = ["A999999999"]
```

Comment:
[Q]
What's the plan here? Listing all scheme ids?

Reply:
As above, we need a business decision on the card schemes we support and then add them on this list.